### PR TITLE
add defaults to connection params

### PR DESCRIPTION
**Pull request by tovganesh**

### CODE REVIEW

File: dbt/adapters/impala/connections.py [hunk 1]
```diff
@@ -36,13 +36,15 @@
 import hashlib
 import threading
 
+DEFAULT_IMAPAL_HOST = "localhost"
```
Comment (sujitkp-blr) [inline dbt/adapters/impala/connections.py:39]: spelling for impala not correct

File: dbt/adapters/impala/connections.py [hunk 2]
```diff
@@ -36,13 +36,15 @@
...
 DEFAULT_IMPALA_PORT = 21050
 DEFAULT_MAX_RETRIES = 3
 
 @dataclass
 class ImpalaCredentials(Credentials):
-    host: str
-    database: Optional[str]
+    host: str = DEFAULT_IMPALA_PORT
```
Comment (sujitkp-blr) [inline dbt/adapters/impala/connections.py:45]: should be DEFUALT_IMPALA_HOST

File: dbt/adapters/impala/connections.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/impala/connections.py]: good catch. fixed.

File: dbt/adapters/impala/connections.py (hunk 2 above)
Comment (tovganesh) [reply dbt/adapters/impala/connections.py]: good catch. fixed.

**Review by sujitkp-blr — APPROVED** [review]

**Pull request merged**